### PR TITLE
perf: create only one `trie` cursor per trie type per merkle execute

### PR DESCRIPTION
**Pull request by joshieDo**
Otherwise, we'd be creating a couple cursors per account, which adds up.

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
lgtm, pending @rakita

File: crates/storage/provider/src/trie/mod.rs [hunk 1]
```diff
@@ -45,9 +46,12 @@ pub enum TrieError {
     UnexpectedCheckpoint,
 }
 
+type AccountsTrieCursor<'tx, TX> =
+    Arc<Mutex<<TX as DbTxMutGAT<'tx>>::CursorMut<tables::AccountsTrie>>>;
```
Comment (gakonst) [inline crates/storage/provider/src/trie/mod.rs:50]: I guess necessary

**Review by gakonst — APPROVED** [review]
fine w me

**Review by rakita — COMMENTED** [review]
lgtm

**Pull request merged**